src/avatars/avatars.controller.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+  Res,
+  StreamableFile,
+  UploadedFile,
+  UseInterceptors,
+} from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { Response } from 'express';
+import * as fs from 'fs';
+import { UploadAvatarRespondDto } from './DTO/upload-avatar.dto';
+import { AvatarsService } from './avatars.service';
+
+@Controller('/avatars')
+export class AvatarsController {
+  constructor(private readonly avatarsService: AvatarsService) {}
+  @Post()
+  @UseInterceptors(FileInterceptor('avatar'))
+  async createAvatar(
+    @UploadedFile() file: Express.Multer.File,
+  ): Promise<UploadAvatarRespondDto> {
+    //const userid = this.authService.verify(auth).userId;
```

Comment:
You should verify auth. Since only logined user can create avatars.

This remind me of a security problem. This function could be attacked by someone with ulterior motives and then create many, many avatars until the server hard disk is overrun. And pruning is only done periodically. So there should be an upload limit even for logged in users as well. But you can open an issue and implemented later.

Reply:
Solve #79 in the future.